source/_posts/2017-01-24-burying-your-dead-code.markdown

```diff
@@ -0,0 +1,31 @@
+---
+layout: post
+title: "Burying Your Dead Code"
+date: 2017-01-24 13:15:12 -0800
+comments: true
+author: "Lindsey Anne <lindsey.hogg@grandrounds.com>"
+categories: git bisect, dead code
+---
+As our code evolves and changes, little bits of leftovers from refactors or changed paths can build up as dead code. It's like the plaque that builds up on your teeth over time: not really noticeable at first, but after a trip to the dentist, you feel a whole lot better.
+
+Removing dead code from your codebase is a simple process at its core: find the dead parts, prove that they're dead, and delete them. The tricky part is in that second step, but after a little trial and error, we discovered a couple tricks to speed up the process.
+
+##### Start small
+
+Before digging into a gigantic repo to see what you can rip out of there, consider the impact of what you're about to do and start with just a small piece of it. Since I was investigating one of our biggest repos, I decided to tackle a single Model first rather than, say, the entire lib directory.
+
+##### Finding the dead parts
+
+I'm sure there's other gems out there that perform similar assessment, but our favorite gem for this is [debride by Seattle Ruby Brigade](https://github.com/seattlerb/debride). It's not 100% perfect, but it will give you a good starting place to start digging around. The whitelisting option is particularly helpful after your initial investigation. Another option is the brute-force method, where you comb through a file and investigate each and every piece of it one by one. If your `debride` output is suspiciously large, this might be worth a try anyway.
```

Comment:
"I'm sure there are other gems ..."

"it will give you a good starting place to start digging around" -> one of those "start"s should probably go.

"each and every piece of it one by one" -> vaguely tautological.

"this might be worth a try anyway" -> the "anyway" here is unnecessary.